src/Router.js

```diff
@@ -114,21 +114,35 @@ export default class Router {
    * @param {Context} ctx
    */
   __runHandlers(handlers, ctx, callback) {
-    let i = 0;
+    let index = 0;
 
     let next = () => {
       if (this.__dispatchId !== ctx.dispatchId) {
         return;
       }
-      let fn = handlers[i]
-      i++
-      // capture i in closure to not fuck
-      var j = i;
-
-      if (fn) {
-        fn(ctx, next)
-        if (callback && j === handlers.length && this.__dispatchId === ctx.dispatchId) {
-          callback();
+      let handlerFnOrArray = handlers[index]
+      index++
+      // capture handler index in closure since index could be modified by another thread
+      const handlerIndex = index;
+
+      if (handlerFnOrArray) {
+        if (Array.isArray(handlerFnOrArray)) {
+          let parallelHandlersComplete = 0
+
+          // for parallel handlers we use a custom next callback that tracks completion of all handlers in the group
+          const parallelNext = () => {
+            parallelHandlersComplete++
+            if (parallelHandlersComplete === handlerFnOrArray.length) {
```

Comment:
nit: could you add a comment here that just clarifies something along the lines of "Since we know this is an array of handlerFns, call the next fn when all are complete?

Just had a small double with the variable name still having "Or" in it and thought something to that effect may help.

Reply:
done, thanks I think that is a good suggestion